### ISSUE TITLE
Redis for Session Storage

### ISSUE THREAD

**Issue by dondragon2** [mrniko/netty-socketio#296]
I and using this the RedisStoreFactory instead of the default.

`
Configuration config = new Configuration();
        config.setHostname("localhost");
        config.setPort(9092);
        config.setStoreFactory(new RedissonStoreFactory());
`

But I am getting the following error once an event comes in.

Caused by: java.lang.AbstractMethodError: com.corundumstudio.socketio.store.RedissonPubSubStore$1.onMessage(Ljava/lang/String;Ljava/lang/Object;)V
	at org.redisson.PubSubMessageListener.onMessage(PubSubMessageListener.java:73) ~[redisson-2.1.3.jar:na]
	at org.redisson.client.RedisPubSubConnection.onMessage(RedisPubSubConnection.java:67) ~[redisson-2.1.3.jar:na]
	at org.redisson.client.handler.CommandDecoder.handleMultiResult(CommandDecoder.java:239) ~[redisson-2.1.3.jar:na]
	at org.redisson.client.handler.CommandDecoder.decodeMulti(CommandDecoder.java:205) ~[redisson-2.1.3.jar:na]
	at org.redisson.client.handler.CommandDecoder.decode(CommandDecoder.java:189) ~[redisson-2.1.3.jar:na]
	at org.redisson.client.handler.CommandDecoder.decode(CommandDecoder.java:97) ~[redisson-2.1.3.jar:na]
	at io.netty.handler.codec.ReplayingDecoder.callDecode(ReplayingDecoder.java:370) ~[netty-codec-4.0.30.Final.jar:4.0.30.Final]

Is this the correct way of using this store method? Can you point to some example or documentation in using this?

**Comment by mrniko** [comment]
@dondragon2 please update Redisson to latest version

**Comment by dondragon2** [comment]
@mrniko I did the update but I think there are some breaking changes when that is used.
com.corundumstudio.socketio.store.RedissonStoreFactory requires a Redission object but Redisson.create(config) produces a RedissonClient object. Also, the following code breaks at compile time.

 public RedissonStoreFactory() {
        this(Redisson.create());
    }

**Comment by dondragon2** [comment]
Ok, After digging deeper I came up with the following. I'm using 'org.redisson:redisson:2.2.3'

```
 Config config = new Config();
        SingleServerConfig singleServerConfig = config.useSingleServer();
        singleServerConfig.setAddress("localhost:6379");

        if (password != null) {
            singleServerConfig.setPassword(password);
        }

        log.info("Creating Redisson config to {} {} password", singleServerConfig.getAddress(), password == null ? "without" : "with");

        Redisson redis = (Redisson) Redisson.create(config);

        Configuration config = new Configuration();
        config.setHostname("localhost");
        config.setPort(9092);
        config.setStoreFactory(new RedissonStoreFactory(redisson));

        final SocketIOServer server = new SocketIOServer(config);
        server.addListeners(new ChatService());
```

But I am still getting the following exception

```
2015-12-21 08:56:41.343  WARN 55083 --- [tLoopGroup-2-12] io.netty.channel.DefaultChannelPipeline  : An exceptionCaught() event was fired, and it reached at the tail of the pipeline. It usually means the last handler in the pipeline did not handle the exception.

io.netty.handler.codec.DecoderException: java.lang.AbstractMethodError: com.corundumstudio.socketio.store.RedissonPubSubStore$1.onMessage(Ljava/lang/String;Ljava/lang/Object;)V
	at io.netty.handler.codec.ReplayingDecoder.callDecode(ReplayingDecoder.java:425) ~[netty-codec-4.0.32.Final.jar:4.0.32.Final]
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:244) ~[netty-codec-4.0.32.Final.jar:4.0.32.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:308) [netty-transport-4.0.32.Final.jar:4.0.32.Final]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:294) [netty-transport-4.0.32.Final.jar:4.0.32.Final]
	at io.netty.channel.ChannelInboundHandlerAdapter.channelRead(ChannelInboundHandlerAdapter.java:86) [netty-transport-4.0.32.Final.jar:4.0.32.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:308) [netty-transport-4.0.32.Final.jar:4.0.32.Final]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:294) [netty-transport-4.0.32.Final.jar:4.0.32.Final]
	at io.netty.channel.ChannelInboundHandlerAdapter.channelRead(ChannelInboundHandlerAdapter.java:86) [netty-transport-4.0.32.Final.jar:4.0.32.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:308) [netty-transport-4.0.32.Final.jar:4.0.32.Final]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:294) [netty-transport-4.0.32.Final.jar:4.0.32.Final]
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:846) [netty-transport-4.0.32.Final.jar:4.0.32.Final]
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:131) [netty-transport-4.0.32.Final.jar:4.0.32.Final]
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:511) [netty-transport-4.0.32.Final.jar:4.0.32.Final]
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:468) [netty-transport-4.0.32.Final.jar:4.0.32.Final]
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:382) [netty-transport-4.0.32.Final.jar:4.0.32.Final]
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:354) [netty-transport-4.0.32.Final.jar:4.0.32.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:112) [netty-common-4.0.32.Final.jar:4.0.32.Final]
	at io.netty.util.concurrent.DefaultThreadFactory$DefaultRunnableDecorator.run(DefaultThreadFactory.java:137) [netty-common-4.0.32.Final.jar:4.0.32.Final]
	at java.lang.Thread.run(Thread.java:745) [na:1.8.0_25]
```

**Comment by mrniko** [comment]
This was my bug. Current version is not compatible with latest version of Redisson. Now it's fixed.

**Comment by dondragon2** [comment]
same issue when using
```
compile 'com.corundumstudio.socketio:netty-socketio:1.7.8'
compile 'org.redisson:redisson:2.2.3'
```

**Comment by mrniko** [comment]
please use netty-socketio from `master` branch. Bug fix is not released yet.

**Comment by dondragon2** [comment]
that worked. thanks.

**Comment by qinerg** [comment]
I use 1.6.5 appear the same fault

**Comment by mrniko** [comment]
@qinerg This issue fixed only for 1.7.x line
